indra/newview/gltf/asset.cpp

```diff
@@ -1083,37 +890,60 @@ void Image::decompose(Asset& asset, const std::string& folder)
         }
         else
         {
+            error = "Unknown mime type, saved as .bin";
             extension = ".bin";
         }
 
-        std::string filename = folder + "/" + name + "." + extension;
+        std::string filename = folder + delim + name + delim + extension;
 
         // set URI to non-j2c file for now, but later we'll want to reference the j2c hash
         mUri = name + "." + extension;
 
         std::ofstream file(filename, std::ios::binary);
         file.write((const char*)buffer.mData.data() + bufferView.mByteOffset, bufferView.mByteLength);
     }
-
-#if 0
-    if (!mData.empty())
+    else if (mTexture.notNull())
     {
-        // save j2c image
-        std::string filename = folder + "/" + name + ".j2c";
-
-        LLPointer<LLImageRaw> raw = new LLImageRaw(mWidth, mHeight, mComponent);
-        U8* data = raw->allocateData();
-        llassert_always(mData.size() == raw->getDataSize());
-        memcpy(data, mData.data(), mData.size());
-
-        LLViewerTextureList::createUploadFile(raw, filename, 4096);
+        auto bitmapmgr = LLLocalBitmapMgr::getInstance();
+        if (bitmapmgr->isLocal(mTexture->getID()))
+        {
+            LLUUID tracking_id = bitmapmgr->getTrackingID(mTexture->getID());
+            if (tracking_id.notNull())
+            { // copy original file to destination folder
+                std::string source = bitmapmgr->getFilename(tracking_id);
```

Comment:
You don't need `LLLocalBitmapMgr` for this. `mTexture` should have a specified url that is `"file://"+mFilename`